lib/util/git/auth.ts

```diff
@@ -138,6 +138,16 @@ export function getAuthenticationRules(
     }/${insteadUrl.full_name}${insteadUrl.git_suffix ? '.git' : ''}`,
   });
 
+  // When on gitlab, we add oauth2 since it's hardcoded.
+  if (token.startsWith('gitlab-ci-token')) {
+    authenticationRules.push({
+      url: url.toString(protocol),
+      insteadOf: `https://oauth2@${insteadUrl.resource}${
+        insteadUrl.port ? `:${insteadUrl.port}` : ''
+      }/${insteadUrl.full_name}${insteadUrl.git_suffix ? '.git' : ''}`,
```

Comment:
you need more test cases for this to cover all branches (conditions).

you can probably compute this part once and reuse it. then you don't need a new test. 😉